feat(lps): Application card filters

ianjon3s opened the pull request:
## What does this PR do?

Sets up search and category filters for LPS application cards.

https://github.com/user-attachments/assets/e66e4259-066f-4066-bf11-4108dfc832f5

DafyddLlyr commented:
Small PR here to address the above issue @ianjon3s - https://github.com/theopensystemslab/planx-new/pull/5270

ianjon3s commented:
@DafyddLlyr I've worked through the steps and all is working as expected:
- Refactored to use `awaitingPayment` following API change
- Removed redundant declaration of `ApplicationCardProps`
- Created new custom hook `useFilters`

ianjon3s commented:
Good call @DafyddLlyr , I've created a reusable checkbox component, also extended input focus styles to work with checkboxes
